### PR TITLE
SINGA-120 - Implemented GRU and BPTT

### PR DESCRIPTION
**Pull request by fanju1984**
This is the pull request for SINGA-120 - Implemented GRU and BPTT

### CODE REVIEW

File: src/neuralnet/neuron_layer/gru.cc [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+		context = &srclayers[1]->data(this);
+	}
+
+	// Compute the update gate
+	GEMM(1.0f, 0.0f, src,*w_z_hx_t,update_gate);
+	if (bias_z_ != nullptr)
+		MVAddRow(1.0f,1.0f,bias_z_->data(),update_gate);
+	Blob<float> zprev (batchsize_,hdim_);
+	GEMM(1.0f, 0.0f, *context,*w_z_hh_t, &zprev);
+	Add<float>(*update_gate, zprev, update_gate);
```
Comment (nudles) [inline src/neuralnet/neuron_layer/gru.cc]: can you explain again why we cannot merge 128 - 130 as 

GEMM(1.0f, 1.0f, *context, &w_z_hh_t, update_gate);

Comment (fanju1984) [inline src/neuralnet/neuron_layer/gru.cc]: I think you can do the merge. 
Previously, I meant we cannot remove line 125, as we need to use the 0.0f to reset the variable `update_gate`

Comment (nudles) [inline src/neuralnet/neuron_layer/gru.cc]: Ok.

**Comment by fanju1984** [comment]
Please also refer to pull request #118

**Pull request closed without merging**